src/osm/GeometryHandler.cpp

```diff
@@ -61,137 +79,402 @@ osm2rdf::osm::GeometryHandler<W>::GeometryHandler(
+  // < r and that point is more than eps away from the line segment connecting l
+  // and r. Then we call the algorithm recursively on the part to the left of m
+  // and the part to the right of m. NOTE: It's a matter of taste whether we
+  // include m in the left recursion or the right recursion, but we should not
+  // include it in both.
+  bool a = innerOuterDouglasPeucker<MODE>(inputPoints, outputPoints, l, m, eps);
+  bool b =
+      innerOuterDouglasPeucker<MODE>(inputPoints, outputPoints, m + 1, r, eps);
+
+  return a || b;
+}
+
+// ____________________________________________________________________________
+template <typename W>
+double GeometryHandler<W>::signedDistanceFromPointToLine(
+    const osm2rdf::geometry::Location& A, const osm2rdf::geometry::Location& B,
+    const osm2rdf::geometry::Location& C) {
+  // Check that the input is OK and not A == B.
+  if (A.get<0>() == B.get<0>() && A.get<1>() == B.get<1>()) {
+    return 0;
+  }
+
+  // The actual computation, see this Wikipedia article for the formula:
+  // https://en.wikipedia.org/wiki/Distance_from_a_point_to_a_line
+  double distAB = sqrt((A.get<0>() - B.get<0>()) * (A.get<0>() - B.get<0>()) +
+                       (A.get<1>() - B.get<1>()) * (A.get<1>() - B.get<1>()));
+  double areaTriangleTimesTwo =
+      (B.get<1>() - A.get<1>()) * (A.get<0>() - C.get<0>()) -
+      (B.get<0>() - A.get<0>()) * (A.get<1>() - C.get<1>());
+  return areaTriangleTimesTwo / distAB;
+}
+
+// ____________________________________________________________________________
+template <typename W>
+osm2rdf::geometry::Area GeometryHandler<W>::simplifiedArea(
+    const osm2rdf::geometry::Area& area, bool inner) const {
+  // default value: empty area, means no inner geom
+  osm2rdf::geometry::Area ret;
+
+  // skip trivial / erroneous geoms
+  if (!boost::geometry::is_valid(area)) {
+    return ret;
+  }
+
+  if (boost::geometry::is_empty(area)) {
+    return ret;
+  }
+
+  size_t numPointsOld = 0;
+  size_t numPointsNew = 0;
+
+  for (const auto& poly : area) {
+    osm2rdf::geometry::Polygon simplified;
+    numPointsOld += poly.outer().size();
+
+    for (const auto& origInner : poly.inners()) {
+      numPointsOld += origInner.size();
+      if (origInner.size() < 4) {
+        numPointsNew += origInner.size();
+        simplified.inners().push_back(origInner);
+        continue;
+      }
+
+      // inner polygons are given in counter-clockswise order
+
+      double eps = sqrt(-boost::geometry::area(origInner) / 3.14) * 3.14 * 2 *
+                   _config.simplifyGeometriesInnerOuter;
+
+      // simplify the inner geometries with outer simplification, because
+      // inner geometries are given counter-clockwise, it is not
+      // necessary to swap the simplification mode
+      boost::geometry::model::ring<osm2rdf::geometry::Location> retDP;
+      size_t m = floor(origInner.size() / 2);
+      if (inner) {
+        innerOuterDouglasPeucker<2>(origInner, retDP, 0, m, eps);
```

Comment:
`enum` for MODE in innerOuterDougasPeucker could improve readability here.

Reply:
I agree,